### PR TITLE
Added training, segmentation, and scoring for ICDAR2015

### PR DESCRIPTION
**Pull request by desh2608**

### CODE REVIEW

**Comment by desh2608** [comment]
@danpovey I'm now running this on the grid. Will probably have the results by tomorrow.

File: egs/icdar2015/v1/=2.2.2 [hunk 1]
```diff
@@ -0,0 +1,19 @@
+Collecting matplotlib
```
Comment (danpovey) [inline egs/icdar2015/v1/=2.2.2]: is this file supposed to be here?

**Comment by desh2608** [comment]
Oh I'll remove it. Sorry!

File: egs/icdar2015/v1/local/eval/script.py [hunk 2]
```diff
@@ -0,0 +1,315 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+from collections import namedtuple
```
Comment (danpovey) [inline egs/icdar2015/v1/local/eval/script.py]: Please have some file-level documentation via a doc-string.

File: egs/icdar2015/v1/local/eval/rrc_evaluation_funcs.py [hunk 3]
```diff
@@ -0,0 +1,366 @@
...
+        if withTranscription and withConfidence:
+            m = re.match(r'^\s*(-?[0-9]+)\s*,\s*(-?[0-9]+)\s*,\s*(-?[0-9]+)\s*,\s*(-?[0-9]+)\s*,\s*(-?[0-9]+)\s*,\s*(-?[0-9]+)\s*,\s*(-?[0-9]+)\s*,\s*(-?[0-9]+)\s*,\s*([0-1].?[0-9]*)\s*,(.*)$',line)
+            if m == None :
+                raise Exception("Format incorrect. Should be: x1,y1,x2,y2,x3,y3,x4,y4,confidence,transcription")
+        elif withConfidence:
+            m = re.match(r'^\s*(-?[0-9]+)\s*,\s*(-?[0-9]+)\s*,\s*(-?[0-9]+)\s*,\s*(-?[0-9]+)\s*,\s*(-?[0-9]+)\s*,\s*(-?[0-9]+)\s*,\s*(-?[0-9]+)\s*,\s*(-?[0-9]+)\s*,\s*([0-1].?[0-9]*)\s*$',line)
+            if m == None :
+                raise Exception("Format incorrect. Should be: x1,y1,x2,y2,x3,y3,x4,y4,confidence")
+        elif withTranscription:
+            m = re.match(r'^\s*(-?[0-9]+)\s*,\s*(-?[0-9]+)\s*,\s*(-?[0-9]+)\s*,\s*(-?[0-9]+)\s*,\s*(-?[0-9]+)\s*,\s*(-?[0-9]+)\s*,\s*(-?[0-9]+)\s*,\s*(-?[0-9]+)\s*,(.*)$',line)
```
Comment (danpovey) [inline egs/icdar2015/v1/local/eval/rrc_evaluation_funcs.py]: You might want to see if it's possible to use regexp notation for repetitions, e.g. `{4}`, to simplify this.  Personally I don't know how but I assume it's doable.

**Comment by aarora8** [comment]
I guess these issues are due to using default evaluation scripts in ICDAR. If it is convenient for you, you can use the scoring scripts from MADCAT setup. It is inspired from ICDAR and should give the same result.

**Comment by danpovey** [comment]
I'm OK to use the scoring code as-is-- just put a note at the top explaining that it is the official ICDAR scoring code and where it came from.  Then it's clear why the style differs from what we use.

**Comment by danpovey** [comment]
oops, there is a conflict now.

**Comment by desh2608** [comment]
Resolved. I think this can be merged now, if the Cuda error is not a code issue?

**Comment by danpovey** [comment]
OK, thanks.  Merging.

**Pull request merged**